docs/JSONDocumentation.md

```diff
@@ -0,0 +1,379 @@
+
+# JSON documentation
+
+This document contains an overview of the possible keys in the intermediary JSON file. It also contains a brief explanation of the meaning of the key in the different programming languages the analyser accepts. Unless indicated otherwise, the terminology used is based on DOTNET.
+
+## Member
+This is the top level element of the file.
+
+NET
+: A member can represent a class, interface, struct or enum. By default, it is a class (if Type is not specified).
+
+Java
+: A member can represent a class, interface, record or enum. By default, it is a class (if Type is not specified).
+
+
+### Required keys:
+
+| Key      | Type   | .NET                                    | Java                                    |
+|----------|--------|-----------------------------------------|-----------------------------------------|
+| FullName | string | Name of the class/interface/struct/enum | Name of the class/interface/record/enum |
+
+### Optional keys:
+
+| Key                   | Type                                             | .NET                                                 | Java                                                |
+|-----------------------|--------------------------------------------------|------------------------------------------------------|-----------------------------------------------------|
+| Type                  | number                                           | 0: class<br/>1: interface<br/>2: struct<br/>3: enum  | 0: class<br/>1: interface<br/>2: record<br/>3: enum |
+| Modifiers             | number                                           | See [Modifiers](#modifiers)                          | See [Modifiers](#modifiers)                         |
+| Basetypes             | array of strings                                 | Base classes                                         | Super classes                                       |
```

Comment:
For both .NET and Java, interfaces also end up in here - that may be unexpected for Java folks, so maybe worthy of mention here.

Reply:
I mean that the key `Basetypes` specifically includes the base class (C#) or superclass (Java), as well as the implemented interfaces (.NET and Java).

In Java, `class Rhombus extends Shape implements Drawable` becomes `"Basetypes": ["Shape", "Drawable"]`. It does not become `"Basetypes": ["Shape"]`.

In C#, the same thing is written `class Rhombus : Shape, IDrawable` (there's not a syntactical difference between extending and implementing) so it's more obvious that these end up in the same array.